src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -55,27 +55,28 @@ internal partial class Binder
 
             if (conversion.IsIdentity)
             {
-                // identity tuple conversions result in a converted tuple
-                // to indicate that tuple conversions are no longer applicable.
-                // nothing else changes
-                if (source.Kind == BoundKind.TupleLiteral)
-                {
-                    var sourceTuple = (BoundTupleLiteral)source;
-                    source = new BoundConvertedTupleLiteral(
-                        sourceTuple.Syntax,
-                        sourceTuple.Type, 
-                        sourceTuple.Arguments, 
-                        sourceTuple.Type, // same type to keep original element names 
-                        sourceTuple.HasErrors);
-                }
-
                 // We need to preserve any conversion that changes the type (even identity conversions, like object->dynamic),
                 // or that was explicitly written in code (so that GetSemanticInfo can find the syntax in the bound tree).
                 if (!isCast && source.Type == destination)
                 {
                     return source;
                 }
             }
+            else if (ReclassifyExpression(ref source, syntax, conversion, isCast && !wasCompilerGenerated, destination, diagnostics))
```

Comment:
In typical code in the vast majority of cases the natural type of the literal will match the type it is passed or assigned to. I do not think we need to reclassify in such cases

```C#
var x = (1,2); // this should not require recursive reclassification

int x = 1;
int y = 2;
foo((x, y));      // this should not require recursive reclassification

void foo<T>(T x)
{
}
```